website/public/js/controllers/partyCtrl.js

```diff
@@ -11,6 +14,20 @@ habitrpg.controller("PartyCtrl", ['$rootScope','$scope','Groups','Chat','User','
 
       if ($state.is('options.social.party')) {
         $scope.group.$syncParty(); // Sync party automatically when navigating to party page
+
+        // Checks if user's party has reached 2 players for the first time.
+        if(!user.achievements.partyUp
+            && $scope.group.memberCount >= 2) {
+          user.achievements.partyUp = true;
```

Comment:
OK, here's the trouble. Simply changing the property in the `user` object here only updates it on the client. You need a `User.set()` call to get it to the server--see, for instance, https://github.com/khipkin/habitrpg/blob/new_party_badges/website/public/js/controllers/partyCtrl.js#L122.